internal/observability/middleware_test.go

```diff
@@ -149,7 +162,7 @@ func TestMiddlewareLogging(t *testing.T) {
 			assert.Equal(t, expected, getLog(), "Unexpected log entry written.")
 		})
 		t.Run(tt.desc+", unary outbound", func(t *testing.T) {
-			res, err := mw.Call(context.Background(), req, fakeOutbound{err: tt.err})
+			res, err := mw.Call(context.Background(), req, fakeOutbound{err: tt.err, applicationErr: tt.applicationErr})
```

Comment:
The construction of the fake handler is getting pretty repetitive. Can we name the `tt` struct and define a small function to hold this repeated code?

```
type case struct {
  desc string
  ...all the other fields...
}

var tests := []case{
  ...cases....
}

newOut := func(c case) fakeOutbound {
  return fakeOutbound{err: case.err, applicationErr: case.applicationErr}
}
```